### PR TITLE
Allowing nesting search query under 'search' key.

### PR DESCRIPTION
**Pull request by ADmad**
This avoid collision with other search options like 'group' etc.

### CODE REVIEW

File: src/Model/Behavior/SearchBehavior.php [hunk 1]
```diff
@@ -45,7 +49,7 @@ public function findSearch(Query $query, array $options)
      * Returns the valid search parameter values according to those that are defined
      * in the searchConfiguration() method of the table.
      *
-     * @param array $param a key value list of search parameters to use for a search.
+     * @param array $params a key value list of search parameters to use for a search.
      * @return array
      */
     public function filterParams($params)
```
Comment (lorenzo) [inline src/Model/Behavior/SearchBehavior.php]: Should this function return the params nested in the the searchKey?

Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: If using the `search` key in options you wouldn't need to use this function.

Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: But will do so for consistency.

Comment (ADmad) [inline src/Model/Behavior/SearchBehavior.php]: Done

**Pull request merged**